lib/errors/createErrorType.js

```diff
@@ -0,0 +1,67 @@
+var util = require("util");
+
+/**
+ * Creates a new type of error.
+ *
+ * @param {string} name - The error type name
+ * @param {string} message - The error message
+ * @param {string[]|function} additionalKeys - Key names to set from arguments (or function to manually do so)
+ *
+ * @returns {function} Returns an Error class
+ *
+ * @private
+ */
+function createErrorType(name, message, additionalKeys) {
+
+    // Create the error type
+    var NewErrorType = function() {
+
+        // Call the "super" constructor
+        Error.call(this);
+
+        // Capture a proper stack trace
+        Error.captureStackTrace(this, this.constructor);
```

Comment:
This is unfortunately a v8-only API. You could feature detect however, and perhaps fall back to
```(new Error()).stack``` as before.

Reply:
Oh, didn't even realize! If there's not any advantage to `.captureStackTrace` I would probably opt to keep it simple with `(new Error()).stack`.

Reply:
It could be useful, just when feature detected. Although, without a repro for the originating issue it's hard to know. Including the constructor stack frame, while annoying isn't a big deal. Including an incorrect stack as in the issue definitely is, but I wonder if it was a REPL only issue.